darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -0,0 +1,213 @@
+import os
+import argparse
+import datetime
+import importlib.resources as pkg_resources
+from typing import Any
+
+import pandas as pd
+from mako.template import Template
+
+import darshan
+import darshan.templates
+
+
+def setup_parser(parser: Any):
+    """
+    Configures the command line arguments.
+
+    Parameters
+    ----------
+    parser : ArgumentParser object
+        Command line argument parser.
+    """
+    parser.add_argument(
+        "log_path",
+        type=str,
+        help="Specify path to darshan log.",
+    )
+    parser.add_argument("--output", type=str, help="Specify output filename.")
+
+
+def get_runtime(report: Any):
+    """
+    Calculates the run time from the report metadata.
+    """
+    # calculate the run time
+    runtime = float(
+        report.metadata["job"]["end_time"] - report.metadata["job"]["start_time"]
+    )
+    if runtime < 1.0:
+        # to prevent the displayed run time from being 0.0 seconds
+        # label anything under 1 second as less than 1
+        runtime = "< 1"
+    return runtime
+
+
+def get_full_command(report: Any):
+    """
+    Retrieves the full command line from the report metadata.
+    """
+    if report.metadata["exe"]:
+        # some logs don't have a stored executable
+        try:
+            # to detect if the stored executable is not an
+            # integer (anonymized) try to convert it to an integer
+            command = int(report.metadata["exe"])
+        except ValueError as err:
+            if "invalid literal" in str(err):
+                # if it cannot be converted to an integer
+                # assign the executable
+                command = report.metadata["exe"]
+        else:
+            # if it can be converted to an
+            # integer label it anonymized
+            command = "Anonymized"
+    else:
+        # if there is no executable
+        # label as not available
+        command = "N/A"
+    return command
+
+
+def get_header(report: Any):
```

Comment:
for all the functions labeled with `report: Any`

I checked locally:

```python
reveal_type(report)
reveal_type(pd.DataFrame) # control for Any
```

`mypy` puts a more specific type on the `report` and labels a `pandas` dataframe as `Any`, so seems like we could be more specific here

```
darshan/cli/summary.py:189: note: Revealed type is "darshan.report.DarshanReport"
darshan/cli/summary.py:190: note: Revealed type is "Any"

Reply:
The latest version has changes all instances of `report: Any` to `report: darshan.report.DarshanReport`, so I believe this can be resolved.